### PR TITLE
Repair pyright in unit test

### LINKED ISSUE #392: Fix types in tests

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Partially resolves #392 

* **What is the current behavior?** (You can also link to an open issue here)

### CODE REVIEW

File: starknet_py/compile/compiler_test.py [hunk 1]
```diff
@@ -31,7 +37,7 @@ def test_compile_direct_load():
 
 def test_compile_file_load():
     output_file_str = Compiler(
-        contract_source=[test_file_path.resolve().absolute()]
+        contract_source=[cast(CairoFilename, test_file_path.resolve().absolute())]
```
Comment (cptartur) [inline starknet_py/compile/compiler_test.py:40]: I'm not sure about cast required for pyright to work with `NewType`s. There should be a solution that doesn't require casting.

File: starknet_py/utils/sync/sync_test.py [hunk 2]
```diff
@@ -76,12 +76,12 @@ async def test_asynchronous_versions():
...
     contract = Contract("1")
-    repository = contract.get_repository_sync()
-    function = repository.get_function_sync()
+    repository = contract.get_repository_sync()  # pyright: ignore
+    function = repository.get_function_sync()  # pyright: ignore
 
     assert function.call_sync() == 1
     assert function.get_name() == "function X"
-    assert contract.example_class_method_sync() == 2
+    assert contract.example_class_method_sync() == 2  # pyright: ignore
```
Comment (cptartur) [inline starknet_py/utils/sync/sync_test.py:84]: Why are ignores needed here?

Comment (war-in) [reply starknet_py/utils/sync/sync_test.py:84]: Because pyright does not recognize _sync methods :/

Comment (cptartur) [reply starknet_py/utils/sync/sync_test.py:84]: I'd consider including some explanations here and in other places, why pyright was disabled. This probably also applies to code already merged

File: starknet_py/compile/compiler_test.py (hunk 1 above)
Comment (war-in) [reply starknet_py/compile/compiler_test.py]: It is already resolved. I've changed CairoFilename from NewType to str 😅

File: starknet_py/net/l1/messaging_test.py [hunk 3]
```diff
@@ -138,7 +140,7 @@ def get_tx_receipt(_tx_hash):
             }
         )
 
-    w3_mock_receipt.eth.get_transaction_receipt = get_tx_receipt
+    w3_mock_receipt.eth.get_transaction_receipt = get_tx_receipt  # pyright: ignore
```
[the remark this answers is not in the document]
Comment (war-in) [reply starknet_py/net/l1/messaging_test.py:142]: Small differences in return types. I don't think it is worth resolving

**Review by Solpatium — APPROVED** [review]

**Pull request merged**